packages/sdk/client-services/src/packlets/invitations/device-invitation-protocol.ts

```diff
@@ -46,6 +47,17 @@ export class DeviceInvitationProtocol implements InvitationProtocol {
     };
   }
 
+  checkInvitation(invitation: Partial<Invitation>) {
+    try {
+      const identity = this._getIdentity();
+      if (identity) {
+        return new AlreadyJoinedError('Currently only one identity per client is supported.');
```

Comment:
returning errors feels kinda odd, why not throw?

Reply:
I started by throwing and was having issues w/ it. In the end I don't think that was the issue but I kept it because throwing seemed to generate a console message even if it was caught? Would you prefer if this was expected to throw if there's a problem and have a return type of `void`?

Reply:
IMO @dmaretskyi 
- throw is something in our code went wrong.
- this is a protocol level expected result so we should return an error.